### ISSUE TITLE
Unable create maven project with latest theia (windows) IDE

### ISSUE THREAD

**Issue by octopus88888** [eclipse-theia/theia#13284]
Scenario 1:
1) Create Maven Project
2) Select spring-boot-blank-archetype
![image](https://github.com/eclipse-theia/theia/assets/142282882/8682a665-580f-441c-8658-4e1e49d38574)

3) Faced error to create maven project
![image](https://github.com/eclipse-theia/theia/assets/142282882/78836dbe-52e6-4aff-ada5-603eb56234c4)

Scenario 2:
Similar error message with command line if provide -DoutputDirectory with invalid path:

mvn archetype:generate -DarchetypeGroupId=com.nokia.dos -DarchetypeArtifactId=mechanism-driver-archetype -DarchetypeVersion=24.1.2 -DoutputDirectory=file:///c%3A/<path>
...
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:3.2.1:generate (default-cli) on project standalone-pom: java.io.IOException: The file
name, directory name, or volume label syntax is incorrect -> [Help 1]

**Comment by JonasHelming** [comment]
Does this only occur with the Spring Boot archetype or with all templates?

**Comment by msujew** [comment]
Since the task tries to execute the `-DoutputDirectory=file:///c%3A/` argument, I assume there's something wrong with out URI decoding/encoding in this part of the API.

**Comment by octopus88888** [comment]
> Does this only occur with the Spring Boot archetype or with all templates?

not only occur with the Spring Boot archetype, tried others also get the same error.

**Comment by sgraband** [comment]
I can take a look!

**Comment by tsmaeder** [comment]
Doesn't this sound like a problem in https://github.com/eclipse-jdtls/eclipse.jdt.ls ?

**Comment by msujew** [comment]
@tsmaeder I haven't confirmed this bug in vscode yet (I don't do Java coding these days), but I assume it works as expected there. In that case it's probably related to Theia in some way or another.

**Comment by sgraband** [comment]
I just checked. The extension (and command) works in VSCode. So i would assume, that the API that the extension uses does not behave the same in Theia. Probably due to the file path decoding/encoding.

**Comment by tsmaeder** [comment]
@sgraband why would `file:///c%3A/<path>` not be the correct url? Looks like the url-encoded form of `c:/<path>` to me.

**Comment by tsmaeder** [comment]
@sgraband can you give more context on "how the extension provides the args for the call"? Is this a json file, an API call?

**Comment by msujew** [comment]
@sgraband I'm not sure whether the file URI is correct or not. It might need to return `uri.path.fsPath` instead. However, we should take a look at what vscode provides there.

**Comment by sgraband** [comment]
> @sgraband why would file:///c%3A/<path> not be the correct url? Looks like the url-encoded form of c:/<path> to me.

@tsmaeder might be, like i said it did not have any impact on executing the task, but i am unsure on what is expected there. I looked at this due to the [comment](https://github.com/eclipse-theia/theia/issues/13284#issuecomment-1898348411) and Issue description above that indicated it might have something to do with this. However like @msujew suggested i can take a look on what vscode provides there.

> can you give more context on "how the extension provides the args for the call"? Is this a json file, an API call?

I took a look at this [function](https://github.com/microsoft/vscode-maven/blob/8ee1ef0bbe3810e96c161d0c0e6c7e72c9c54490/src/archetype/ArchetypeModule.ts#L154). 
I mainly debugged the `commandLine` values from above at [this line](url).

**Comment by tsmaeder** [comment]
@sgraband the last url is broken.

**Comment by sgraband** [comment]
Sorry about that: https://github.com/eclipse-theia/theia/blob/3e7f6eec297c4029ef01b2ef2c8f20361360c48c/packages/task/src/node/process/process-task-runner.ts#L88 (also fixed it above)

**Comment by tsmaeder** [comment]
No, sorry, @sgraband I have no idea what the right thing to do is, here. In general, it's a good idea to keep things as objects instead of encoding them into a string for as long as possible, but not sure how that applies to this problem.

**Comment by john-brice** [comment]
This is still an issue with 1.61.0
Any Maven archetype results in the same error.
